### PR TITLE
Metrics properly watch with new delimiter

### PR DESCRIPTION
**Pull request by mlajkim**
# Description

The current internal metrics correctly monitors role certificates for the delimiter `:role.`, but fails to do so for other delimiters like `_`, for example.

## Type of change

- [X] Bug fix
- [ ] New feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Non-code changes (update documentation, pipeline, etc.)

### Flags

- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code

---

## Checklist

- [x] Followed the guidelines in the CONTRIBUTING document
- [x] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [x] Tested and linted the code
- [x] Commented the code
- [x] Made corresponding changes to the documentation
- [x] Passed all pipeline checking

## Checklist for maintainer
- [x] Use `Squash and merge`
- [x] Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- [x] Delete the branch after merge

### CODE REVIEW

File: pkg/identity/identity.go [hunk 1]
```diff
@@ -407,12 +407,10 @@ func PrepareRoleCsrOptions(cfg *config.IdentityConfig, domain, service string) (
 	for _, domainrole := range strings.Split(cfg.TargetDomainRoles, ",") {
 		// referred to SplitRoleName()
 		// https://github.com/AthenZ/athenz/blob/73b25572656f289cce501b4c2fe78f86656082e7/libs/go/sia/util/util.go#L69-L78
-		dr := strings.Split(domainrole, ":role.")
-		if len(dr) != 2 || len(dr[0]) == 0 || len(dr[1]) == 0 {
-			return nil, fmt.Errorf("Invalid role name: '%s', expected format {domain}:role.{role}", domainrole)
+		targetDomain, targetRole, err := extutil.DomainRoleSplitter(domainrole, ":role.")
+		if err != nil {
+			continue
```
Comment (mlajkim) [inline pkg/identity/identity.go:412]: Oops. Should return error here.

Comment (mlajkim) [reply pkg/identity/identity.go]: handled.

File: pkg/identity/metricsd.go [hunk 2]
```diff
@@ -72,7 +73,14 @@ func Metricsd(idConfig *config.IdentityConfig, stopChan <-chan struct{}) (error,
 
 	if idConfig.TargetDomainRoles != "" && idConfig.RoleCertDir != "" {
 		for _, domainrole := range strings.Split(idConfig.TargetDomainRoles, ",") {
-			exporter.Files = append(exporter.Files, strings.TrimSuffix(idConfig.RoleCertDir, "/")+"/"+domainrole+".cert.pem")
+			targetDomain, targetRole, err := athenz.SplitRoleName(domainrole)
+			if err != nil {
+				continue
+			}
+			// if RoleCertFilenameDelimiter = "_" then,
+			// fileName = your-domain_your-role.cert.pem"
```
Comment (WindzCUHK) [inline pkg/identity/metricsd.go:81]:
```suggestion
```
can remove the comment?

File: pkg/identity/metricsd.go [hunk 3]
```diff
@@ -72,7 +73,14 @@ func Metricsd(idConfig *config.IdentityConfig, stopChan <-chan struct{}) (error,
 
 	if idConfig.TargetDomainRoles != "" && idConfig.RoleCertDir != "" {
 		for _, domainrole := range strings.Split(idConfig.TargetDomainRoles, ",") {
-			exporter.Files = append(exporter.Files, strings.TrimSuffix(idConfig.RoleCertDir, "/")+"/"+domainrole+".cert.pem")
+			targetDomain, targetRole, err := athenz.SplitRoleName(domainrole)
+			if err != nil {
+				continue
```
Comment (WindzCUHK) [inline pkg/identity/metricsd.go:78]:
```suggestion
				log.Warnf("Failed to initialize metrics exporter with TARGET_DOMAIN_ROLES: %s, err: %s", idConfig.TargetDomainRoles, err.Error())
				continue
```

Comment (mlajkim) [reply pkg/identity/metricsd.go:78]: @WindzCUHK 
Let me not include "failed to" as it simply skips with `continue` for individual malicious element of target_domain_role

File: pkg/identity/metricsd.go (hunk 2 above)
Comment (mlajkim) [reply pkg/identity/metricsd.go:82]: You are right, a bit superfluous. but then I think it could be helpful for those who are not familiar with Athenz.

Comment (mlajkim) [reply pkg/identity/metricsd.go:82]: nvm. will delete it.

File: pkg/identity/metricsd.go (hunk 3 above)
Comment (WindzCUHK) [reply pkg/identity/metricsd.go:79]: maybe "fail to monitor ... role", plx help to fix the wordings

**Review by WindzCUHK — APPROVED** [review]

**Pull request merged**